src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/RepoFetchingSkyKeyComputeState.java

```diff
@@ -71,6 +74,13 @@ enum Signal {
   // we might block in `close()`, which is potentially bad (see its javadoc).
   @Nullable volatile Future<RepositoryDirectoryValue.Builder> workerFuture = null;
 
+  /** The executor service that manages the worker thread. */
+  // We hold on to this alongside `workerFuture` because it's the only reliable way to make sure the
+  // worker thread has shut down (the `Future` class doesn't have the capability).
+  final ExecutorService workerExecutorService =
+      Executors.newThreadPerTaskExecutor(
+          Profiler.instance().profileableVirtualThreadFactory("starlark-repository-"));
```

Comment:
nit: maybe add the name of the repository being fetched to the name of the thread?